main.cpp

```diff
@@ -4991,11 +5001,37 @@ bool encrypt_command::execute(device_map &devices) {
         fail(ERROR_ARGS, "Can only sign to same file type");
     }
 
-    if (get_file_type_idx(2) != filetype::bin) {
+    if (!settings.filenames[2].empty() && settings.filenames[2].find("0x") == 0) {
+        // Hex string instead of file
```

Comment:
Feels like this "parse a long hex string into an array of byes" probably ought to be abstracted out to a separate function, to avoid code-repetition?